### DISCUSSION TITLE
Context Propagation over D-Bus Inter Process Communication

### DISCUSSION THREAD

**Discussion by utezduyar** [open-telemetry/community#1299] [Community Discussions]
D-Bus (DBus) is a pretty standard way of doing IPC in the Linux world. It uses linux IPC primitives to provide Application level APIs. The long but valuable specification of it is here https://dbus.freedesktop.org/doc/dbus-specification.html

I was wondering if there are any experimentations done by any of you to do context propagation over DBus. DBus has built in data type primitives. I am not sure what the project thinks (haven't discussed it with them yet) but maybe a standard way of context propagation in built to the DBus protocol might be very beneficial. 

Hope to read your comments.

**Comment by dyladan** [comment]
Sorry for the late reply here. I'm not aware of any experimental propagation over DBus. Is there any particular application you're hoping to instrument which would require this or are you just generally curious?

**Comment by utezduyar** [comment]
Thanks for your reply. I am curious in general. We use dbus heavily in our product and it would be really good to trace the calls. I have started some discussions with upstream dbus community. It can be traced here - https://lists.freedesktop.org/archives/dbus/2023-January/018229.html
